testing/test_segment_volume.py

```diff
@@ -0,0 +1,110 @@
+                        "sub-test001_T1w_seg-manual.nii.gz")
+BATCH_SIZE = 1
+DROPOUT = 0.4
+BN = 0.1
+LENGTH_3D = [96, 96, 16]
+
+
+def test_segment_volume_2d():
+    model = imed_models.Unet(in_channel=1,
+                             out_channel=1,
+                             depth=2,
+                             drop_rate=DROPOUT,
+                             bn_momentum=BN)
+
+    # temporary folder that will be deleted at the end of the test
+    if not os.path.exists(PATH_MODEL):
+        os.mkdir(PATH_MODEL)
+
+    torch.save(model, os.path.join(PATH_MODEL, "model_test.pt"))
+    config = {
+        "batch_size": BATCH_SIZE,
+
+        "transformation_validation": {
+            "Resample": {"wspace": 0.75, "hspace": 0.75},
+            "ROICrop2D": {"size": [48, 48]},
+            "ToTensor": {},
+            "NormalizeInstance": {}
+        },
+        "slice_filter": {
+            "filter_empty_mask": False,
+            "filter_empty_input": False
+        },
+        "slice_filter_roi": 10,
+        "slice_axis": "axial",
+        "unet_3D": False,
+    }
+    PATH_CONFIG = os.path.join(PATH_MODEL, 'model_test.json')
+    with open(PATH_CONFIG, 'w') as fp:
+        json.dump(config, fp)
+
+    nib_img = imed_utils.segment_volume(PATH_MODEL, IMAGE_PATH, ROI_PATH)
+
+    assert nib_img.shape == nib.load(IMAGE_PATH).shape
+    assert (nib_img.dataobj.max() < 1.0) and (nib_img.dataobj.max() > 0.0)
+    assert nib_img.dataobj.dtype == 'float32'
+
+    shutil.rmtree(PATH_MODEL)
+
+
+def test_segment_volume_3d():
+
+    model = imed_models.UNet3D(in_channels=1,
+                               n_classes=1,
+                               base_n_filter=1)
+
+    # temporary folder that will be deleted at the end of the test
+    if not os.path.exists(PATH_MODEL):
+        os.mkdir(PATH_MODEL)
+
+    torch.save(model, os.path.join(PATH_MODEL, "model_test.pt"))
+    config = {
+        "batch_size": BATCH_SIZE,
+        "transformation_validation": {
+            "Resample": {"wspace": 1.0, "hspace": 1.0, "dspace": 2.0},
+            "CenterCrop3D": {"size": LENGTH_3D},
+            "ToTensor3D": {},
+            "NormalizeInstance3D": {}
+        },
+        "slice_filter": {
+            "filter_empty_mask": False,
+            "filter_empty_input": False
+        },
+        "slice_filter_roi": None,
+        "slice_axis": "sagittal",
+        "unet_3D": True,
+        "length_3D": LENGTH_3D,
+        "padding_3D": 0,
+    }
+
+    PATH_CONFIG = os.path.join(PATH_MODEL, 'model_test.json')
+    with open(PATH_CONFIG, 'w') as fp:
+        json.dump(config, fp)
+
+    nib_img = imed_utils.segment_volume(PATH_MODEL, IMAGE_PATH)
+    assert nib_img.get_fdata()[..., 0].shape == nib.load(IMAGE_PATH).shape
+    assert (nib_img.dataobj.max() < 1.0) and (nib_img.dataobj.max() > 0.0)
+    assert nib_img.dataobj.dtype == 'float32'
+
+    shutil.rmtree(PATH_MODEL)
+
+
+print("Test segment volume with 2D model.")
+test_segment_volume_2d()
```

Comment:
i'm coming a bit late (PR already merged) but why calling this function here? With pytest (which [seems to be what we are using](https://github.com/neuropoly/ivado-medical-imaging/blob/master/requirements.txt#L8)), all tests are run if they are defined with `def test_my_function` (which already the case), without the need to call them explicitly.

Reply:
It's indeed not required, it was this way on our other tests + it is easier to debug with an IDE (simply run file), but it is not very elegant. I'll change the tests in the next PR to remove those explicit calls.

Reply:
yes, i suggest to remove them as it is confusing (at least it confused me)-- plus: you can run pytest within most IDE (e.g. Pycharm's debug mode allows you to run nose, pytest, etc.): Run > Edit configuration... > template > Python tests > pytest